File: src/Dto/Project.Dto.Http/Position/PositionDto.cs
namespace Project.Dto.Http.Position;

public class PositionDto
{
    public PositionDto(Guid id, Guid? parentId, string title, Guid companyId, bool isDeleted)
    {
        Id = id;
        ParentId = parentId;
        Title = title;
        CompanyId = companyId;
        IsDeleted = isDeleted;
    }

    public Guid Id { get; init; }
    public Guid? ParentId { get; init; }
    public string Title { get; init; }
    public Guid CompanyId { get; init; }
    
    public bool IsDeleted { get; init; }
}

File: front/src/app/dto/position.dto.ts
/**
 * DTO for position received from API
 */
export interface PositionDto {
  id: string;
  parentId?: string;
  title: string;
  companyId: string;
  isDeleted: boolean;
}

/**
 * DTO for position hierarchy received from API
 */
export interface PositionHierarchyDto {
  positionId: string;
  parentId?: string;
  title: string;
  level: number;
}

/**
 * DTO for creating a position
 */
export interface CreatePositionDto {
  title: string;
  companyId: string;
  parentId?: string;
}

/**
 * DTO for updating a position
 */
export interface UpdatePositionDto {
  title?: string;
  parentId?: string;
}

